idaes/apps/grid_integration/tests/test_backcaster.py

```diff
@@ -0,0 +1,192 @@
+#################################################################################
+# The Institute for the Design of Advanced Energy Systems Integrated Platform
+# Framework (IDAES IP) was produced under the DOE Institute for the
+# Design of Advanced Energy Systems (IDAES), and is copyright (c) 2018-2021
+# by the software owners: The Regents of the University of California, through
+# Lawrence Berkeley National Laboratory,  National Technology & Engineering
+# Solutions of Sandia, LLC, Carnegie Mellon University, West Virginia University
+# Research Corporation, et al.  All rights reserved.
+#
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and
+# license information.
+#################################################################################
+
+import pytest
+from pyomo.common import unittest as pyo_unittest
+from idaes.apps.grid_integration.forecaster import ForecastError, Backcaster
+
+
+@pytest.fixture
+def historical_da_prices():
+    return {"test_bus": [1] * 24 + [2] * 24 + [3] * 24}
+
+
+@pytest.fixture
+def historical_rt_prices():
+    return {"test_bus": [10] * 24 + [20] * 24 + [30] * 24}
+
+
+@pytest.fixture
+def base_backcaster(historical_da_prices, historical_rt_prices):
+    return Backcaster(historical_da_prices, historical_rt_prices)
+
+
+@pytest.mark.unit
+def test_create_backcaster(historical_da_prices, historical_rt_prices):
+    backcaster = Backcaster(historical_da_prices, historical_rt_prices)
+    assert backcaster.historical_da_prices is historical_da_prices
+    assert backcaster.historical_rt_prices is historical_rt_prices
+
+
+@pytest.mark.unit
+def test_create_backcaster_with_small_max_historical_days(
+    historical_da_prices, historical_rt_prices
+):
+    max_n_days = 1.0
+    with pytest.raises(Warning, match=r".*Dropping the first day's data.*"):
+        backcaster = Backcaster(
+            historical_da_prices, historical_rt_prices, max_historical_days=max_n_days
+        )
+        expected_historical_da_prices = {"test_bus": [3] * 24}
+        expected_historical_rt_prices = {"test_bus": [30] * 24}
+
+        pyo_unittest.assertStructuredAlmostEqual(
+            first=backcaster.historical_da_prices, second=expected_historical_da_prices
+        )
+
+        pyo_unittest.assertStructuredAlmostEqual(
+            first=backcaster.historical_rt_prices, second=expected_historical_rt_prices
+        )
+
+
+@pytest.mark.unit
+@pytest.mark.parametrize("value", ["10", "ten", [10]])
+def test_invalid_max_historical_days_type(
+    value, historical_da_prices, historical_rt_prices
+):
+    with pytest.raises(TypeError, match=r".*max_historical_days must be a number.*"):
```

Comment:
Change the error message to `...max_historical_days must be a positive integer`?

Reply:
I came up with a way to solve this issue. Instead of "raise warning", I use "warnings.warn", and in test_backcaster.py, add one line:
pytestmark = pytest.mark.filterwarnings("error")

It seems this line of code can convert warning to error when testing. 

I have pushed the codes to my repo.